### PR TITLE
OpenFAST v3.2.0

### PR DESCRIPTION
**Pull request by rafmudaf**
**Feature or improvement description**
Pull request to merge `main` into `dev` and create a tagged release for `v3.2.0`.

See the milestone and project pages for additional information
- https://github.com/OpenFAST/openfast/milestone/4
- https://github.com/OpenFAST/openfast/projects/5 for more information.

**Test results, if applicable**
See GitHub Actions

# Release Checklist

- [x] Update the documentation version in docs/conf.py
- [x] Update the versions in docs/source/user/api_change.rst
- [x] Verify readthedocs builds correctly
- [x] Create a tag in OpenFAST
- [x] Create a merge commit in r-test and add a corresponding tag
- [x] Compile executables for Windows builds
    - [x] FAST_SFunc.mexw64
    - [x] OpenFAST-Simulink_x64.dll
    - [x] openfast_x64.exe
    - [x] DISCON.dll

# Proposed Change Log

### General
#866 Include legacy documentation in pdf and MS Word format - Modules
#1021 Reg_tests python scripts: fix issue where directory returned is empty
#1062 Add documentation for community contribution
#1169 Update git-module urls
#1158 Update orientation differences in linear trim solution
#1177 Add documentation on 3D rotations in linearization
#1178 Corrections for 3.2.0 release

### AeroDyn
#932 BugFix: UA update states that were not updated
#1032 AD: added more nodal outputs for debugging UA
#1039 [BugFix] Nacelle motion not passed between AeroDyn driver and AeroDyn
#1043 AeroDyn: combine some FVW and BEMT output calculations
#1045 OLAF: check for division by zero to avoid invalid calculations
#1141 Move AD module reg tests to a standalone job

### BeamDyn
#1050 Fix aeroelastic stability analysis with BeamDyn

### FAST.Farm
#1107 Fix Time-Step Delay in Super Controller within FAST.Farm

### HydroDyn
#806 C-bindings interface for HydroDyn
#1047 Fix HydroDyn Jacobian outputs when LinOutJac is True
#1108 BugFix: Checks to populate matrix input to Newman's app were reversed
#1173 Bug Fix: rotation matrix perturbation with small angles was wrong

### InflowWind
#1144 BugFix: Fix typo in DLLEXPORT attributes for IfW C interface

### MAP++
#1048 MAP: small fix based on MAP 1.3
#1168 Fix MAP linearization operating point

### NWTC Library
#1050 Fix aeroelastic stability analysis with BeamDyn -- changes to the library routines for angle perturbations during linearization
#1124 Updates to VersionInfo module and its dependencies
#1157 Fix for binary file compression

### OpenFAST Library
#962 Add a Python glue-code interface
#1057 FAST Library: add access to hub position and velocity

### ServoDyn
#803 Linearization of ServoDyn Structural control elements
#1074 [BugFix] ServoDyn StC control signal channels were not zeroed properly
#1089 BugFix: Fix API change line numbers
#1101 BugFix: ServoDyn API change docs
#1140 Add non-rotating hub forces to Bladed Interface
#1160 Correction to documentation on Extended Bladed DLL interface

### TurbSim
#887 TurbSim modifications

### CODE REVIEW

**Review by ebranlard — APPROVED** [review]
I would possibly add #1168

**Comment by bjonkman** [comment]
I'd suggest also adding this bug fix: https://github.com/OpenFAST/openfast/pull/1158

**Review by andrew-platt — APPROVED** [review]

File: docs/source/user/api_change.rst [hunk 1]
```diff
@@ -46,6 +58,7 @@ OpenFAST                                      28   Patm            103500     Pa
 OpenFAST                                      29   Pvap            1700       Pvap              - Vapour pressure of working fluid (Pa) [used only for an MHK turbine cavitation check]
 OpenFAST                                      30   WtrDpth         50         WtrDpth           - Water depth (m)
 OpenFAST                                      31   MSL2SWL         0          MSL2SWL           - Offset between still-water level and mean sea level (m) [positive upward]
+TurbSim                                       13   WrHAWCFF        False      WrHAWCFF          - Output full-field time-series data in HAWC form?  (Generates RootName-u.bin, RootName-v.bin, RootName-w.bin, RootName.hawc)
```
Comment (jjonkman) [inline docs/source/user/api_change.rst:61]: WrHAWCFF is a new input added in OpenFAST v3.2.0, and should not show up in the v3.1.0 table.

File: docs/source/user/fast.farm/index.rst [hunk 2]
```diff
@@ -1,15 +1,14 @@
-.. _FAST.Farm:
 
-FAST.Farm User’s Guide and Theory Manual
+FAST.Farm User's Guide and Theory Manual
 ========================================
 
-.. only:: html
+This document offers a quick reference guide for the FAST.Farm driver for
```
Comment (jjonkman) [inline docs/source/user/fast.farm/index.rst:5]: The sentence, "This document offers a quick reference guide for the FAST.Farm driver for OpenFAST." should be deleted.

File: docs/source/user/turbsim/output.rst [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Notes: 
+
+1. The ``factor_scaling`` values in the summary file indicate the inverse of the values TurbSim used to scale the data in the HAWC files.
+   ``factor_scaling`` can theoretically be used in HAWC2 to obtain the original data generated by TurbSim.
+   TurbSim scales the data so that HAWC2 will obtain the standard-deviation ratios of 1.0 (u/u), 0.8 (v/u), and 0.5 (w/u) as a work-around for
+   an issue in how HAWC2 scales turbulence files. Please note that these ratios may not work well with non-IEC turbulence models.
+
+2. HAWC-formatted files are always periodic, so all of the analysis time steps are written.
+   
+3. The u-component wind speed files have the mean *hub-height* wind speed removed, so they will contain any shear that was defined.
```
Comment (jjonkman) [inline docs/source/user/turbsim/output.rst:25]: Presumably you mean "The u-component wind speed files have the mean *hub-height* wind speed removed; they will contain any shear that was defined minus this mean."?

**Review by jjonkman — CHANGES_REQUESTED** [review]
I'm happy to see OpenFAST v3.2.0 finally being released!  Please find a few comments regarding the updated documentation below.

Also, after finally reviewing #1050 that has already been merged, I found a couple discrepancies wrt the 3D rotation implementation plan.  I've now sent these comments to @andrew-platt and will paste them here for completeness.  These changes should also be implemented in this release:

- For the perturbation of input rotations for the TStc and SStc meshes in ServoDyn, the small angle perturbation approach should be used, i.e., the call to PerturbOrientationMatrix() needs UseSmlAngle=.true. For BStc and NStc, the implementation looks correct.

- Likewise in AeroDyn for the tower motion orientation, UseSmlAng = .true. should be used.

File: docs/source/user/turbsim/output.rst (hunk 3 above)
Comment (andrew-platt) [reply docs/source/user/turbsim/output.rst:25]: See #1178 for correction

File: docs/source/user/fast.farm/index.rst (hunk 2 above)
Comment (andrew-platt) [reply docs/source/user/fast.farm/index.rst:5]: See #1178 for correction

File: docs/source/user/api_change.rst (hunk 1 above)
Comment (andrew-platt) [reply docs/source/user/api_change.rst:61]: See #1178 for correction

File: docs/source/user/api_change.rst [hunk 4]
```diff
@@ -10,10 +10,22 @@ The line number corresponds to the resulting line number after all changes are i
...
+
+None
+
 
-No change
+
+OpenFAST v3.1.0 to OpenFAST v3.2.0
+----------------------------------
+============================================= ==== =============== ========================================================================================================================================================================================================
+Removed in OpenFAST dev
```
[the remark this answers is not in the document]
Comment (andrew-platt) [reply docs/source/user/api_change.rst:23]: See #1178 for correction

**Review by jjonkman — APPROVED** [review]
Thanks for addressing my comments.  I know approve this release.

**Pull request merged**